presto-main/src/main/java/com/facebook/presto/operator/aggregation/AbstractStatisticalDigestAggregationFunction.java

```diff
@@ -0,0 +1,220 @@
+        Type intermediateType = stateSerializer.getSerializedType();
+        StatisticalDigestStateFactory statisticalFactory;
+        if (tdigest) {
+            statisticalFactory = StatisticalDigestStateFactory.createTDigestFactory();
+        }
+        else {
+            statisticalFactory = StatisticalDigestStateFactory.createQuantileDigestFactory();
+        }
+
+        AggregationMetadata metadata = new AggregationMetadata(
+                generateAggregationName(name, outputType.getTypeSignature(), inputTypes.stream().map(Type::getTypeSignature).collect(toImmutableList())),
+                createInputParameterMetadata(inputTypes),
+                getMethodHandle(valueType, arity),
+                COMBINE_FUNCTION,
+                OUTPUT_FUNCTION.bindTo(stateSerializer),
+                ImmutableList.of(new AccumulatorStateDescriptor(
+                        StatisticalDigestState.class,
+                        stateSerializer,
+                        statisticalFactory)),
+                outputType);
+
+        GenericAccumulatorFactoryBinder factory = AccumulatorCompiler.generateAccumulatorFactoryBinder(metadata, classLoader);
+        return new InternalAggregationFunction(name, inputTypes, ImmutableList.of(intermediateType), outputType, true, true, factory);
+    }
+
+    private static List<Type> getInputTypes(Type valueType, int arity)
+    {
+        switch (arity) {
+            case 1:
+                // weight and accuracy unspecified
+                return ImmutableList.of(valueType);
+            case 2:
+                // weight specified, accuracy unspecified
+                return ImmutableList.of(valueType, BIGINT);
+            case 3:
+                // weight and accuracy specified
+                return ImmutableList.of(valueType, BIGINT, DOUBLE);
+            default:
+                throw new IllegalArgumentException(format("Unsupported number of arguments: %s", arity));
+        }
+    }
+
+    private MethodHandle getMethodHandle(Type valueType, int arity)
+    {
+        final MethodHandle inputFunction;
+        switch (valueType.getDisplayName()) {
+            case StandardTypes.DOUBLE:
+                if (tdigest) {
+                    inputFunction = INPUT_DOUBLE_T;
+                }
+                else {
+                    inputFunction = INPUT_DOUBLE_Q;
+                }
+                break;
+            case StandardTypes.REAL:
+                if (tdigest) {
+                    throw new UnsupportedOperationException();
+                }
+                inputFunction = INPUT_REAL;
+                break;
+            case StandardTypes.BIGINT:
+                if (tdigest) {
+                    throw new UnsupportedOperationException();
+                }
+                inputFunction = INPUT_BIGINT;
+                break;
+            default:
+                throw new IllegalArgumentException(format("Unsupported type %s supplied", valueType.getDisplayName()));
```

Comment:
Can we change these to `PrestoException`?  This was a pre-existing bug.  If we throw IAE then it shows up as a Presto bug which definitely isn't the case.